main.c

```diff
@@ -144,43 +184,57 @@ static uint32_t MAIN_MPU_330(uint32_t port, uint32_t val, uint32_t out)
           c = '\n';
           mpu_dbg_ctr = 0;
         }
-        DBG_Log("%02x%c", val, c);
+        DBG_Logi("%02x%c", val, c);
```

Comment:
Yes, a hex dump of the raw MIDI data. I think it is a nice feature.

Reply:
Yes I understand. So it is NOT needed for release builds?
You can add one more specific macro like _LOG() but without file/lines, which will be disabled in release build. Or I can add that and you may use it, we can do that later.

Reply:
It IS intended for release builds.
if you use the option /MDBG2 you can output the MIDI data over the debug serial port at 115200 baud, 3 times the MIDI serial rate of 38400 baud (different from normal MIDI which is 31500 baud.)
You can dump the actual MIDI data that games are using while the music actually being played, without having to intercept it physically.

Reply:
OK, cool :)

Reply:
Yes, it is OK to print to the screen if /MDBG is specified but /DBG is not.
BTW, the hexdump uses 3 characters for each MIDI data byte, two for the MIDI data byte, and one more for the space or newline. Ignoring the LF to CRLF conversion in serial.c

BTW, is there a good reason to have the LF to CRLF conversion and the baud rate set to 9600?
In my builds I remove the LF to CRLF conversion, since in my Linux terminal using cat or minicom I get extra blank lines with CRLF.
And 9600 is just too slow so I set it to 115200.

Reply:
It might depend on the terminal, for me it needs CRLF to properly turn to next line.
I don't know why the baud rate is set, you may ask @thp who should know about this.

Reply:
It's probably fine to have 115200 and LF only. As @jiyunomegami mentioned, it's up to the terminal on the other side to be configured properly.